storage/innobase/fil/fil0crypt.cc

```diff
@@ -1504,22 +1510,21 @@ static bool fil_crypt_find_space_to_rotate(
 
 	state->space = fil_space_t::next(state->space, *recheck,
 					 key_state->key_version != 0);
+	
+	if (srv_read_only_mode) {
+		state->space->release();
+		state->space = fil_system.space_list.end();
+		fil_crypt_return_iops(state,true);
+		return false;
+	}
```

Comment:
It would be simpler to avoid creating any encryption threads if `high_level_read_only` holds. This change basically allows the encryption threads to exist and to waste CPU cycles for achieving absolutely nothing.